server/authentication_test.go

```diff
@@ -42,83 +43,70 @@ func TestSSLEnforcement(t *testing.T) {
 	defer leaktest.AfterTest(t)
 	s := StartTestServer(t)
 	defer s.Stop()
+
+	// HTTPS with client certs.
+	certsContext := testutils.NewRootTestBaseContext()
+	// HTTPS without client certs.
+	noCertsContext := testutils.NewRootTestBaseContext()
+	noCertsContext.Certs = ""
+	// Plain http.
+	insecureContext := testutils.NewRootTestBaseContext()
+	insecureContext.Insecure = true
+
 	testCases := []struct {
-		method, key   string
-		certsStatus   int // Status code for https with client certs.
-		noCertsStatus int // Status code for https without client certs.
+		method, key string
+		ctx         *base.Context
+		success     bool // request sent successfully (may be non-200)
+		code        int  // http response code
 	}{
 		// /ui/: basic file server: no auth.
```

Comment:
no, that's correct, only the KV requests have authentication applied.
the insecureContext attempts to speak HTTP to an HTTPS server, which fails. however, client certificates are not checked, hence the "no auth" comment.